mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/applicationgraph/ApplicationGraphCreator.java

```diff
@@ -103,17 +99,21 @@ private MessageProcessor convertToComponent(Element xmlElement, Flow parentFlow,
       } else {
         report.report("nocompatibility.dynamicflowref", xmlElement, xmlElement);
       }
-    } else if (isOperation(xmlElement)) {
-      return new MessageOperation(xmlElement, parentFlow);
+    } else if (isPropertySource(xmlElement, parentFlow)) {
+      if (isFirstElement(xmlElement, parentFlow)) {
+        return new MessageSource(xmlElement, parentFlow);
```

Comment:
Being the first element in the flow does not necessary mean that it is a message source.
A flow without source could have an `http:request` as the first element for instance.

Reply:
I just removed the 2 classes (MessageSource and MessageOperation) and created a single implementation